src/app/model/ChessEngine.ts

```diff
@@ -67,4 +72,49 @@ export class ChessEngine implements ChessModel {
   private changeTurn(side: Side): Side {
     return side === Side.WHITE ? Side.BLACK : Side.WHITE;
   }
+
+  private willBeKingChecked(player: Player, chessboard: Chessboard, squareFrom: Square, squareTo: Square): boolean {
+    let isCheckedSquareFlag = false;
+
+    const proposedChessboard = _.cloneDeep(chessboard);
```

Comment:
Bardzo OK, ze znalazłeś rozwiązanie na clone deep :) Ale włąśnie Ci powiem, czemu normalnie dąże do niemutowalności (np. chessBoarda). Gdybysmy po kazdym ruchu powiedzmy zwracali nowy board, to wtedy nie ma tego problemu. Bo nie musisz robic kopi, tylko robisz sobie nowa wersje w takich "predykcjach" :) Ale w przypadku naszego zalozenia, ze modyfikujemy chessBoard przy ruchach jest OK.

Reply:
O ile na backendzie nie ma to zbytnio znaczenia, to na froncie kazda dodana biblioteka zwieksza wage aplikacji. Dasz rade to zrobic bez lodasha :) ? Powinno to byc bardzo proste. Ale to tak jako trening, niekoniecznie.

Reply:
Mogę, widziałem 7 różnych sposobów, ale tylko ten wydał się sposobem, który najbardziej mnie zadowalał.
Dzięki o info o wadze bibliotek. Dobrze wiedzieć. A czy przypadkiem parcel nie pobierze końcowo tylko tego kodu, który bedzie nam potrzebny do naszego projektu, a resztę dodatków wytnie?

Widzę też, że pomieszałem opis zadania. Spytałem się chłopaków, jak to zrobić najlepiej (a nie pokazałem im opisu), no i zapomniałem o tym opisie.

Reply:
Tutaj kopiuję, ze względu na to, że muszę zasymulować daną pozycję króla, która to pozycja może być nielegalna. Dlatego tez ruch na docelowej szachownicy byłby niewskazany